### PR TITLE
New Transaction Format

### PR DESCRIPTION
**Pull request by jannikluhn**
This adds the sharding transaction class. Initially I tried to copy and adapt `vm/forks/frontier/*` as well, but at some point I realized that this is useless as everything would have to be rewritten as part of the account abstraction PR anyway. So this PR remains fairly basic.

One test is x-failing because of a bug in pyRLP, I'm gonna fix that over there.

### CODE REVIEW

**Comment by jannikluhn** [comment]
Wait, that's not a bug in pyRLP, I'm just confused by too many brackets. Will fix tomorrow.

File: evm/rlp/sedes.py [hunk 1]
```diff
@@ -1,11 +1,52 @@
...
+)
+
+
+class AccessListElement(CountableList):
+
+    def __init__(self):
+        super().__init__(Binary(max_length=32))
+
+    def serialize(self, obj):
+        result = super().serialize(obj)
```
Comment (pipermerriam) [inline evm/rlp/sedes.py]: Can we move this to below the `if/else` so that it only happens *if* the validation checks pass.

Comment (jannikluhn) [reply evm/rlp/sedes.py]: I'd prefer to leave it there, as `super().serialize(obj)` does a lot of checks already. In particular, that `obj` is actually a list and `obj[0]` is binary. If that weren't the case, the error messages would be misleading.

Comment (pipermerriam) [reply evm/rlp/sedes.py]: That's fine.  I wasn't aware of the  checks that occur in the super call.

**Comment by pipermerriam** [comment]
:shipit:

**Pull request merged**